src/nrniv/CMakeLists.txt

```diff
@@ -260,7 +260,9 @@ endif()
 if(NRN_ENABLE_MPI)
   if(NRN_ENABLE_MPI_DYNAMIC)
     add_library(nrnmpi_lib SHARED ${NRN_NRNMPI_SRC_FILES})
-    target_link_libraries(nrnmpi_lib ${MPI_C_LIBRARIES})
+    if(LINK_AGAINST_MPI)
```

Comment:
I dint the previous behavior strange. If we want to be dynamic, whats the sense of fixed linking?

Reply:
that's fair point. What neuron internally does is:
* dlopen libmpi.so then libmpich.so
* dlopen libnrnmpi.so

This would work if MPI libraries names are standard and LD_LIBRARY_PATH is correctly set. If `libnrnmpi.so` has a rpath to correct `libmpi*.so` then LD_LIBRARY_PATH etc. is not needed. And hence I kept the old behaviour.

Reply:
The #491 multimpi branch resolves these issues. In particular, for linux an mac, the libnrnmpi_xxx no longer links
explicitly to libmpi. I believe this works because libmpi is dlopen'd prior to libnrnmpi_xxx.